src/main/java/com/aws/greengrass/shadowmanager/PubSubIntegrator.java

```diff
@@ -167,4 +176,15 @@ private void handlePublishedMessage(PublishEvent publishEvent) {
                     .log("Unable to perform shadow operation");
         }
     }
+
+    /**
+     * Checks if the message received is a response message or not.
+     *
+     * @param topic the topic on which the PubSub message was sent.
+     * @return true if the message is a response message to a shadow operation; Else false.
+     */
+    private boolean isResponseMessage(String topic) {
+        final Matcher matcher = shadowResponsePattern.matcher(topic);
+        return matcher.find() && matcher.groupCount() == 5;
```

Comment:
why check the group count? This just tells you how many groups are in the pattern, not whether they all matched.

e.g.:
```
$aws/things/foo/shadow/name/bar/update/rejected
``` 
has values for all 5 groups

```
$aws/things/foo/shadow/update/rejected
```
has null values for groups 2 and 3

Is it enough to just do `return matcher.matches()`?